File: TA5/tool-evaluation-4/HAMR/examples/test_event_port_periodic_domains/CAmkES_seL4/src/main/component/test_event_port_periodic_domains/test_event_port_periodic_domains/consumer_t_impl_dst_process_dst_thread.scala
// #Sireum

package test_event_port_periodic_domains.test_event_port_periodic_domains

import org.sireum._
import test_event_port_periodic_domains._

// This file will not be overwritten so is safe to edit
object consumer_t_impl_dst_process_dst_thread {

  def initialise(api: consumer_t_impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

  }

  def timeTriggered(api: consumer_t_impl_Operational_Api): Unit = {
    // example api usage

    val apiUsage_consume: Option[art.Empty] = api.get_consume()
    api.logInfo(s"Received on consume: ${apiUsage_consume}")
  }

  def activate(api: consumer_t_impl_Operational_Api): Unit = { }

  def deactivate(api: consumer_t_impl_Operational_Api): Unit = { }

  def finalise(api: consumer_t_impl_Operational_Api): Unit = { }

  def recover(api: consumer_t_impl_Operational_Api): Unit = { }
}

File: TA5/tool-evaluation-4/HAMR/examples/test_event_port_periodic_domains/CAmkES_seL4/src/c/ext-c/consumer_t_impl_dst_process_dst_thread/consumer_t_impl_dst_process_dst_thread.c
#include <consumer_t_impl_dst_process_dst_thread_api.h>
#include <consumer_t_impl_dst_process_dst_thread.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "consumer_t_impl_dst_process_dst_thread.c", "", "test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_initialise_", 0);

  // examples of api setter and logging usage

  api_logInfo__test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF string("Example logInfo"));

  api_logDebug__test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF string("Example logDebug"));

  api_logError__test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF string("Example logError"));
}

Unit test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "consumer_t_impl_dst_process_dst_thread.c", "", "test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_finalise_", 0);
}

Unit test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_timeTriggered_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "consumer_t_impl_dst_process_dst_thread.c", "", "test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread_timeTriggered_", 0);

  // examples of api getter usage

  if(api_get_consume__test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF_LAST )){
    String consume_str = string("Received event on consume");
    api_logInfo__test_event_port_periodic_domains_test_event_port_periodic_domains_consumer_t_impl_dst_process_dst_thread(SF consume_str);
  }
}
